ppl/cmd/zident/user/new.go

```diff
@@ -0,0 +1,91 @@
+package user
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+
+	"github.com/brimsec/zq/ppl/cmd/zident/root"
+	"github.com/mccanne/charm"
+	"gopkg.in/auth0.v5"
+	"gopkg.in/auth0.v5/management"
+)
+
+var NewUser = &charm.Spec{
+	Name:  "new",
+	Usage: "zident user new",
+	Short: "create new user",
+	Long: `
+Creates a new user in Auth0.
+
+See 'zident help user' for required authentication.
+`,
+	New: NewNewCommand,
+}
+
+type NewCommand struct {
+	*root.Command
+	a0cfg         auth0ClientConfig
+	email         string
+	emailVerified bool
+	tenantID      string
+}
+
+func NewNewCommand(parent charm.Command, f *flag.FlagSet) (charm.Command, error) {
+	c := &NewCommand{Command: parent.(*Command).Command}
+	f.StringVar(&c.email, "email", "", "email address of new user")
+	f.StringVar(&c.tenantID, "tenantid", "", "tenant_id to set instead of creating one")
+	return c, nil
+}
+
+func (c *NewCommand) Run(_ []string) error {
+	if err := c.a0cfg.FromEnv(); err != nil {
+		return err
+	}
+	if c.email == "" {
+		return errors.New("must specify an email address")
+	}
+
+	var tenantID string
+	if c.tenantID != "" {
+		if err := validTenantID(c.tenantID); err != nil {
+			return err
+		}
+		tenantID = c.tenantID
+	} else {
+		tenantID = newTenantID()
+	}
+
+	userID := newUserID()
+
+	m, err := management.New(c.a0cfg.domain.String(),
+		management.WithClientCredentials(c.a0cfg.clientId, c.a0cfg.clientSecret))
```

Comment:
This is our vendor secret from Auth0, right? How does c.a0cfg, which contains the secrets, become available to this command?